main.go

```diff
@@ -228,9 +230,20 @@ func main() {
 			http.NotFound(w, req)
 			return
 		}
-		ok := auth.Handle(w, req)
-		if !ok {
-			return
+
+		ignorePathFound := false
+		for _, path := range cfg.ignorePaths {
+			if req.URL.Path == path {
+				ignorePathFound = true
+				break
+			}
+		}
+
+		if !ignorePathFound {
+			ok := auth.Handle(w, req)
+			if !ok {
+				return
+			}
```

Comment:
When the user configures both `--ignore-paths` and `--allow-paths` it can get quite confusing:

case 1: if `--ignore-paths` is a subset of `--allow-paths`, simply some or all of the allowed paths will be ignored.
case 2: if `--ignore-paths` is not a subset of `--allow-paths` then the former setting has no effect.

Instead, to have clarity for the user, wouldn't it make more sense to have those settings mutually exclusive?

Reply:
Yeah I agree complex behaviour and a potential cause of confusion. I've updated the PR to check that only one of these flags can be used.